### PR TITLE
Cat 2484  refacturing drone, adding animation brick for JumpingSumo drone

### PR DESCRIPTION
**Pull request by cpozvek**

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/content/actions/JumpingSumoAnimationAction.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+public class JumpingSumoAnimationAction extends TemporalAction {
+
+	private ARDeviceController deviceController;
+	private JumpingSumoDeviceController controller;
+
+	private JumpingSumoAnimationsBrick.Animation animationEnum;
+	private float duration;
+
+	public void setAnimationEnum(JumpingSumoAnimationsBrick.Animation animationEnum) {
+		this.animationEnum = animationEnum;
```
Comment (cheidenreich) [inline catroid/src/main/java/org/catrobat/catroid/content/actions/JumpingSumoAnimationAction.java]: please rename to animationType, also the name of the method

File: catroid/src/main/java/org/catrobat/catroid/content/actions/JumpingSumoAnimationAction.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+		animation();
+	}
+
+	protected void animation() {
+
+		if (deviceController != null) {
+			switch (animationEnum) {
+				case SPIN:
+					deviceController.getFeatureJumpingSumo().sendAnimationsSimpleAnimation(ARCOMMANDS_JUMPINGSUMO_ANIMATIONS_SIMPLEANIMATION_ID_ENUM.ARCOMMANDS_JUMPINGSUMO_ANIMATIONS_SIMPLEANIMATION_ID_SPIN);
```
Comment (cheidenreich) [inline catroid/src/main/java/org/catrobat/catroid/content/actions/JumpingSumoAnimationAction.java]: in general avoid the word "enum" in names, datatypes should not be part of it

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/JumpingSumoAnimationsBrick.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	private String animation;
+	private transient Animation animationenum;
+
+	public enum Animation {
+		SPIN, TAB, SLOWSHAKE, METRONOME, ONDULATION, SPINJUMP, SPIRAL, SLALOM
+	}
+
+	public JumpingSumoAnimationsBrick(Animation animation) {
+		this.animationenum = animation;
+		this.animation = animationenum.name();
```
Comment (cheidenreich) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/JumpingSumoAnimationsBrick.java]: here you can call rename the `animationenum` to `animation`
and `animation` to `animationName`

**Comment by cheidenreich** [comment]
please run espresso tests

**Comment by cpozvek** [comment]
can be merged now

**Pull request merged**